### PR TITLE
feat: Added discord provider

### PR DESCRIPTION
**Pull request by sakthiRathinam**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: docs/docs/providers/documentation/discord.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+sidebar_label: Discord Provider
+---
+
+# Discord
+
+:::note Brief Description
+Template Provider is a template for newly added provider's documentation
```
Comment (talboren) [inline docs/docs/providers/documentation/discord.md:8]:
```suggestion
Discord provider is a provider that allows to send notifications to Discord
```

File: docs/docs/providers/documentation/discord.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- components: list[dict] = []: Adding styling or interactive components like emoji,buttons
+
+## Outputs
+*No information yet, feel free to contribute it using the "Edit this page" link the bottom of the page*
+
+## Authentication Parameters
+The `webhook_url` associated with the channel requires to trigger the message to the respective channel.
+
+## Connecting with the Provider
+*No information yet, feel free to contribute it using the "Edit this page" link the bottom of the page*
```
Comment (talboren) [inline docs/docs/providers/documentation/discord.md:22]: Perhaps add a small instruction on how to get that webhook_url?

File: docs/docs/providers/getting-started.md [hunk 3]
```diff
@@ -14,5 +14,5 @@ Common providers include:
 -   **Cloud**: AWS, GCP, Azure, etc.
 -   **Monitoring**: Sentry, New Relic, Datadog, etc.
 -   **Incident Management**: PagerDuty, OpsGenie, etc.
--   **Communication**: Email, Slack, Console, etc.
+-   **Communication**: Email, Slack, Console, Discord etc.
```
Comment (talboren) [inline docs/docs/providers/getting-started.md:17]:
```suggestion
-   **Communication**: Email, Slack, Discord, etc.
```

File: keep/providers/discord_provider/discor_provider.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
+"""
+DiscordOutput is a class that implements the BaseOutputProvider interface for Discord messages.
```
Comment (talboren) [inline keep/providers/discord_provider/discor_provider.py:2]:
```suggestion
DiscordProvider is a class that implements the BaseOutputProvider interface for Discord messages.
```

File: keep/providers/discord_provider/discor_provider.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        """
+        Notify alert message to Discord using the Discord Incoming Webhook API
+        https://discord.com/developers/docs/resources/webhook
+
+        Args:
+            kwargs (dict): The providers with context
+        """
+        self.logger.debug("Notifying alert message to Discord")
+        webhook_url = self.authentication_config.webhook_url
+        content = kwargs.pop("content", "")
```
Comment (talboren) [inline keep/providers/discord_provider/discor_provider.py:47]: Perhaps add an exception if both the content and components are empty?

File: keep/providers/discord_provider/discor_provider.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        components = kwargs.pop("components", [])
+        response = requests.post(
+            webhook_url,
+            json={"content": content, "components": components},
+            headers={
+                "Content-Type": "application/json"
+            }
+        )
```
Comment (talboren) [inline keep/providers/discord_provider/discor_provider.py:55]: When using `json=` it's redundant
```suggestion
        response = requests.post(
            webhook_url,
            json={"content": content, "components": components}
        )
```

File: pyproject.toml [hunk 7]
```diff
@@ -35,6 +35,8 @@ pre-commit-hooks = "^4.4.0"
 yamllint = "^1.29.0"
 black = "^23.1.0"
 isort = "^5.12.0"
+autopep8 = "^2.0.1"
+flake8 = "^6.0.0"
```
Comment (talboren) [inline pyproject.toml:39]: Why are these required? Perhaps remove them from the dependencies?

**Review by talboren — CHANGES_REQUESTED** [review]
Hey @sakthiRathinam thank you so much for your contribution! happy to have you here!

I'll be glad to help you push forward your provider to the main branch.
I left a few comments, please let me know what you think and we can progress with this :).

File: keep/providers/discord_provider/discor_provider.py (hunk 5 above)
Comment (sakthiRathinam) [reply keep/providers/discord_provider/discor_provider.py:47]: added

File: pyproject.toml (hunk 7 above)
Comment (sakthiRathinam) [reply pyproject.toml:39]: For autoformatting and autolinting this was my favourites and always in my vscode settings

**Review by talboren — APPROVED** [review]
LGTM! 🥳
